vault/token_store.go

```diff
@@ -2234,6 +2234,16 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
 			renewable = false
 		}
 
+		// Update num_uses which is equal to req.Data["num_uses"] at this point
+		// 0 means unlimited so 1 is actually less than 0
+		switch {
+		case role.TokenNumUses > 0 && te.NumUses == 0:
+			te.NumUses = role.TokenNumUses
+		case role.TokenNumUses < te.NumUses:
```

Comment:
If TokenNumUses is 0 and te.NumUses is 2, this will actually end up setting te.NumUses to 0. As a base case, if role.TokenNumUses is zero, it should probably just use whatever te.NumUses is.

Reply:
Yes. Oops.
[This test case](https://github.com/hashicorp/vault/pull/6701/files#diff-31edd64725bd3bdf2e3211c5cb9e8fddR4109) catches that and passes with the new base case.